src/lua/log.lua

```diff
@@ -544,9 +426,15 @@ local function check_cfg(cfg)
         end
     end
 
+    local log_type = ffi.C.log_type()
     -- Parsing for validation purposes
     if cfg.log ~= nil then
-        parse_format(cfg.log)
+        log_type = parse_format(cfg.log)
+    end
+
+    if log_type == ffi.C.SAY_LOGGER_SYSLOG and cfg.format == 'json' then
+        box.error(box.error.CFG, 'log format',
+                  "'json' can't be used with syslog logger")
```

Comment:
I don't like that now we use `ER_CFG`, which was original thrown only on `box.cfg` errors, in the log module. Aesthetically, this doesn't look good to me, but it may be all right. @alyapunov WDYT?

Reply:
In terms of PR this allows to reduce code. And in terms of meaning this is just same configuration error just like in case of box.

Reply:
It will be really painful to throw box.error or error depending on calling context.